cpp/ql/lib/semmle/code/cpp/ir/dataflow/internal/SsaInternals.qll

```diff
@@ -256,23 +257,63 @@ private predicate sourceVariableHasBaseAndIndex(SourceVariable v, BaseSourceVari
 }
 
 abstract class DefImpl extends DefOrUseImpl {
-  Operand address;
   int ind;
 
   bindingset[ind]
   DefImpl() { any() }
 
+  override int getIndirectionIndex() { result = ind }
+
+  override string toString() { result = "Def of " + this.getSourceVariable() }
+
   abstract int getIndirection();
 
+  abstract predicate isCertain();
+
   abstract Node0Impl getValue();
+}
 
-  abstract predicate isCertain();
+/** An initial definition of an `IRVariable`'s address. */
+private class DefAddressImpl extends DefImpl, TDefAddressImpl {
+  BaseIRVariable v;
 
-  Operand getAddressOperand() { result = address }
+  DefAddressImpl() {
+    this = TDefAddressImpl(v) and
+    ind = 0
```

Comment:
Since we're defining the SSA definition of the pointer itself. We already have SSA definitions for everything these point to, but the whole point of this PR is that we're added SSA definitions of the pointer itself. So if we changed it to `ind = 1` we'd simply be providing a definition of what the pointer points to (which we already have)